paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/sink/cdc/CdcRecordStoreMultiWriteOperatorTest.java

```diff
@@ -0,0 +1,696 @@
+                                "")));
+    }
+
+    private void createTestTables(Catalog catalog, Tuple2<Identifier, Schema>... tableSpecs)
+            throws Exception {
+
+        for (Tuple2<Identifier, Schema> spec : tableSpecs) {
+            catalog.createTable(spec.f0, spec.f1, false);
+        }
+    }
+
+    @AfterEach
+    public void after() {
+        // assert all connections are closed
+        Predicate<Path> pathPredicate = path -> path.toString().contains(tempDir.toString());
+        assertThat(TraceableFileIO.openInputStreams(pathPredicate)).isEmpty();
+        assertThat(TraceableFileIO.openOutputStreams(pathPredicate)).isEmpty();
+    }
+
+    private static FileIO getFileIO(CatalogContext catalogContext, Path warehouse) {
+        FileIO fileIO;
+        try {
+            fileIO = FileIO.get(warehouse, catalogContext);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        return fileIO;
+    }
+
+    @Test
+    @Timeout(30)
+    public void testAsyncTableCreate() throws Exception {
+        // the async table will have same row type, partitions, and pks as firstTable
+        Identifier tableId = Identifier.create(databaseName, "async_new_table");
+
+        OneInputStreamOperatorTestHarness<CdcMultiplexRecord, MultiTableCommittable> harness =
+                createTestHarness(catalogLoader);
+
+        harness.open();
+
+        Runner runner = new Runner(harness);
+        Thread t = new Thread(runner);
+        t.start();
+
+        // check that records should be processed after table is created
+        Map<String, String> fields = new HashMap<>();
+        fields.put("pt", "0");
+        fields.put("k", "1");
+        fields.put("v", "10");
+
+        CdcMultiplexRecord expected =
+                CdcMultiplexRecord.fromCdcRecord(
+                        databaseName,
+                        tableId.getObjectName(),
+                        new CdcRecord(RowKind.INSERT, fields));
+        runner.offer(expected);
+        CdcMultiplexRecord actual = runner.poll(1);
+
+        assertThat(actual).isNull();
+
+        catalog.createTable(tableId, firstTableSchema, true);
+        actual = runner.take();
+        assertThat(actual).isEqualTo(expected);
+
+        // after table is created, record should be processed immediately
+        fields = new HashMap<>();
+        fields.put("pt", "0");
+        fields.put("k", "3");
+        fields.put("v", "30");
+        expected =
+                CdcMultiplexRecord.fromCdcRecord(
+                        databaseName,
+                        tableId.getObjectName(),
+                        new CdcRecord(RowKind.INSERT, fields));
+        runner.offer(expected);
+        actual = runner.take();
+        assertThat(actual).isEqualTo(expected);
+
+        runner.stop();
+        t.join();
+        harness.close();
+    }
+
+    @Test
+    @Timeout(30)
+    public void testInitializeState() throws Exception {
```

Comment:
I mean using `OneInputStreamOperatorTestHarness.initializeState` to test state recover for the operator.